components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ResourceDeploymentIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+        MockEndpoint.assertIsSatisfied(context);
+        if (!mock.getExchanges().isEmpty()) {
+            Exchange exchange = mock.getExchanges().get(0);
+            Object body = exchange.getIn().getBody();
+            assertTrue(body instanceof ProcessDeploymentResponse);
+            assertTrue(((ProcessDeploymentResponse) body).isSuccess());
+        }
+
+        // Deploy with resource name in header and resource as String in body
+        template.sendBodyAndHeader("direct:deployResource",
+                new String(this.getClass().getClassLoader().getResourceAsStream(RESOURCE_PATH + RESOURCE_NAME).readAllBytes()),
+                ZeebeConstants.RESOURCE_NAME, RESOURCE_NAME);
+        MockEndpoint.assertIsSatisfied(context);
+        if (!mock.getExchanges().isEmpty()) {
+            Exchange exchange = mock.getExchanges().get(0);
+            Object body = exchange.getIn().getBody();
+            assertTrue(body instanceof ProcessDeploymentResponse);
+            assertTrue(((ProcessDeploymentResponse) body).isSuccess());
+        }
+    }
+
+    @Test
+    void testUploadProcessDefinitionJSON() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:deployResource_JSON");
+        mock.expectedMinimumMessageCount(1);
+
+        DeploymentRequest deploymentRequest = new DeploymentRequest();
+        deploymentRequest.setName(RESOURCE_NAME);
+        deploymentRequest
+                .setContent(this.getClass().getClassLoader().getResourceAsStream("data/test1_definition.bpmn").readAllBytes());
+
+        template.sendBody("direct:deployResource_JSON", objectMapper.writeValueAsString(deploymentRequest));
+        MockEndpoint.assertIsSatisfied(context);
+        if (!mock.getExchanges().isEmpty()) {
+            Exchange exchange = mock.getExchanges().get(0);
+            String body = exchange.getIn().getBody(String.class);
+            ProcessDeploymentResponse response = objectMapper.readValue(body, ProcessDeploymentResponse.class);
+            assertTrue(response.isSuccess());
+        }
+    }
+
+    @Test
+    void testInvalidUploadProcessDefinition() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:deployResource");
+        mock.expectedMinimumMessageCount(1);
+        mock.expectedHeaderReceived(ZeebeConstants.IS_SUCCESS, false);
+
+        DeploymentRequest deploymentRequest = new DeploymentRequest();
+        deploymentRequest.setName(INVALID_RESOURCE_NAME);
+        deploymentRequest
+                .setContent(this.getClass().getClassLoader().getResourceAsStream(RESOURCE_PATH + RESOURCE_NAME).readAllBytes());
+
+        template.sendBody("direct:deployResource", deploymentRequest);
+        MockEndpoint.assertIsSatisfied(context);
+        if (!mock.getExchanges().isEmpty()) {
+            Exchange exchange = mock.getExchanges().get(0);
+            Object body = exchange.getIn().getBody();
+            assertTrue(body instanceof DeploymentResponse);
```

Comment:
IIRC, there's an assert specific to check for the type.

Reply:
assertObjectIsClass does not exist in JUnit5. It seems it was removed. Stackoverflow recommends instances